### PR TITLE
Add test for negative dash offset

### PR DESCRIPTION
**Pull request by sagudev**
repro for https://github.com/linebender/vello/issues/1066

### CODE REVIEW

**Comment by Hywan** [comment]
Now https://github.com/linebender/kurbo/pull/454 is merged, can we merge this one?

**Comment by sagudev** [comment]
The test is currently set for old kurbo, so it expects a panic. When we will have updated kurbo I will make this into regression test.

**Review by tomcur — DISMISSED** [review]
LGTM (once the test passes)!

File: vello_tests/tests/regression.rs [hunk 1]
```diff
@@ -89,3 +89,26 @@ fn text_stroke_width_zero() {
...
+    );
+    let mut params = TestParams::new("negative_dash_offset", 30, 30);
+    params.anti_aliasing = AaConfig::Area;
+    smoke_snapshot_test_sync(scene, &params)
+        .unwrap()
+        .assert_mean_less_than(if cfg!(target_os = "macos") {
+            0.005
+        } else {
+            0.001
+        });
```
Comment (tomcur) [inline vello_tests/tests/regression.rs:113]: Do you know why there is a difference between macOS and other systems here? Is it because of FastMath, or is it because macOS is tested on the GPU in CI?

Comment (sagudev) [reply vello_tests/tests/regression.rs:113]: IDK, I do not own mac system to check this.
